### PR TITLE
bubble sort

### PR DESCRIPTION
**Pull request by EricMbouwe**
## SORTING ARRAY
   
  In this project we used: 
  - Methods,
  - loops
  - conditionals

### CODE REVIEW

File: bubble_sort_by.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    (0...arr.length - 1).each do |i|
+      if (arr[i].length - arr[i + 1].length).positive?
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        sorted = true
+      end
+    end
+    break unless sorted
+  end
+  arr
+end
```
Comment (euqueme) [inline bubble_sort_by.rb]: Project Specification (**requirements from Odin**)

* [ ] bubble_sort_by method is implemented and working ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))

- bubble_sort_by has to use yield and only works if a block is given if not it should retrieve an error your's returns the unsorted array when no block is given

image

- also, when trying to test with an array of integers is retrieving a NoMethodError so let's fix it!:
image

**Review by euqueme — CHANGES_REQUESTED** [review]
>### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @EricMbouwe and @abrormukimov :lollipop: 

### Good job, your repo is professional and your bubble_sort method works properly so this is a great start :medal_sports: :medal_sports:  you still have to work on your bubble_sort_by method :construction: :construction_worker_man: :construction_worker_man:  so please address the comments I've left in your code bellow :point_down: 

### That's it, team! :raised_hands: 

### Please :pray: do the changes :man_technologist: :man_technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any question? ask me on slack **@ Maru**

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::::Require changes🛑
# 👋 team,
### Good job on addressing the issue raised in the previous review.
* [ ] You need to enable linter check on this repo. You can visit stickler-ci to enable linters.
> Note: having linters set up and enabled is a mandatory requirement for every project.
Linters config: Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))

### That is it kindly make the change and submit the project for review.
#### Happy coding💻
Addo

**Review by stratospherique — APPROVED** [review]
### Great work implementing the suggested changes :clap: :clap: The project is approved :heavy_check_mark: You may merge :+1: :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request merged**